### DISCUSSION TITLE
Release Planning

### DISCUSSION THREAD

**Discussion by johmsalas** [johmsalas/text-case.nvim#103] [General]
**v1.0.0**
The objective of this milestone is to confirm the plugin's stability by introducing automated tests for its fundamental features

**v2.0.0**
Goal: To make the user experience smoother by adding features that enrich daily interactions

**Comment by peterfication** [comment]
Great idea :)

I would be thrilled to announce it! :) I will need to think about a proper way to do it then, e.g. some screencasts for the different features.

If we start with versioning, I would propose to add a changelog.

Let's try out Github Projects for that. It's a good opportunity, even if it might be overkill for the amount of tickets that need to be handled. Will you create a board and add the issues there that should be closed before v1? Then I will add what I think is missing (if something is missing).

**Reply by johmsalas** [reply]
Added release-please for automatic changelog
Looks promising: https://github.com/johmsalas/text-case.nvim/pull/105

TIL: 
https://github.com/googleapis/release-please#how-should-i-write-my-commits
https://www.conventionalcommits.org/en/v1.0.0/

**Reply by johmsalas** [reply]
Also created the Projects for release plan 1.0.0 and Next. Let me know if you could see them

**Reply by peterfication** [reply]
Yes, I can see the projects :)

**Reply by peterfication** [reply]
I create a milestone as well: https://github.com/johmsalas/text-case.nvim/milestone/1

It somehow integrates nicer in Github issues than projects I have the feeling.

**Reply by johmsalas** [reply]
Time to celebrate! 🥂 Cheers!!

https://github.com/johmsalas/text-case.nvim/milestone/1

**Reply by peterfication** [reply]
Yes! 🎉🎉🎉

I'm working on the release text and screen recordings.

**Comment by johmsalas** [comment]
**Goals for v2.0.0**
- To make the user experience smoother by adding features that enrich daily interactions

Proposing a goal for the v2. features that we are already building: text-case-start-replacing, text-case-start-replacing-with-first-part-command, multicursor

**Reply by peterfication** [reply]
If we follow semantic versioning, we also have to break some things to satisfy a major release :D

**Reply by peterfication** [reply]
I like the goal :)
